File: frontend/src/types/progress.ts
export interface StatusBreakdown {
  new: number;
  learning: number;
  known: number;
  total: number;
}

export interface VocabularyProgress {
  by_status: StatusBreakdown;
  total_successful_recalls: number;
  total_failed_recalls: number;
  total_times_seen: number;
  items_practised: number;
}

export interface CharacterProgress {
  by_status: StatusBreakdown;
  total_successful_recognitions: number;
  total_failed_recognitions: number;
  total_times_seen: number;
  items_practised: number;
}

export interface ProgressStatsResponse {
  vocabulary: VocabularyProgress;
  characters: CharacterProgress;
}

File: backend/src/chinese_learning/presentation/rest/schemas/progress.py
"""Schemas for progress stats API."""

from pydantic import BaseModel


class StatusBreakdownSchema(BaseModel):
    new: int
    learning: int
    known: int
    total: int


class VocabularyProgressSchema(BaseModel):
    by_status: StatusBreakdownSchema
    total_successful_recalls: int
    total_failed_recalls: int
    total_times_seen: int
    items_practised: int


class CharacterProgressSchema(BaseModel):
    by_status: StatusBreakdownSchema
    total_successful_recognitions: int
    total_failed_recognitions: int
    total_times_seen: int
    items_practised: int


class ProgressStatsResponse(BaseModel):
    vocabulary: VocabularyProgressSchema
    characters: CharacterProgressSchema
